### DISCUSSION TITLE
HC620 API source?

### DISCUSSION THREAD

**Discussion by brianmichalk1** [zwave-js/zwave-js#6799] [Q&A]
I did a modified factory reset on an HC620, and cannot get it into pairing mode.  Where is the documentation on the API for this lock?  Is there any docs on LED or beep codes?  Any help is appreciated to find more than the elementary pairing instructions.

**Comment by AlCalzone** [comment]
Try excluding it again even though it isn't part of the network. Afterwards, inclusion might just work again.

I always Google for `<device name> manual pdf` to find the device manual.

**Reply by brianmichalk1** [reply]
Oh, I have plenty of manuals, including the original that came with the lock.  I did find the API documentation here
```
https://products.z-wavealliance.org/ProductManual/File?folder=&filename=product_documents/4279/SB700BehaviorGuide_v2_02.pdf
```
At least it tells me what the modified reset is, but still no information on beep and LED codes.  The idiot manual only covers the most basic of operations.

By the way, thanks for an amazing tool.  I just started using it a couple weeks ago after years of fighting with Openhab's implementation.

I'm about to put together another node to bridge buildings, and the only thing I see I wish was there is an option to route logging over MQTT, or through a remote logging socket.

**Reply by AlCalzone** [reply]
> At least it tells me what the modified reset is, but still no information on beep and LED codes

I'm actually sitting in a room with the firmware engineer right now. I'll ask him later where to find that info.

> the only thing I see I wish was there is an option to route logging over MQTT, or through a remote logging socket.

There is a way to do that. The downside is that you have to provide the transport implementation yourself: https://zwave-js.github.io/node-zwave-js/#/usage/log-transports
...and you have to enable it in Z-Wave JS UI somehow, unless you're using node-zwave-js barebones. There may even be a way to do logging over websocket already, since Home Assistant and Z-Wave JS UI both have logging in their frontend. @robertsLando may be able to tell you how to do both.

**Reply by brianmichalk1** [reply]
Man, I would love to know how they come up with their design document.  Not that it's bad, but how do they determine what features to implement?

**Reply by robertsLando** [reply]
@brianmichalk1 There is actually no documentation about ZUI websockets but you could just connect to the websocket and listen for `DEBUG` events. In order to connect to websocket here is the code I use:

https://github.com/zwave-js/zwave-js-ui/blob/c37ae2ad368013715bf894403703208d170cc756/src/App.vue#L991-L999

If auth is enabled you firstly need a token that you can get with a POST request to `/api/authenticate` with a JSON with `username` and `password` (defautl to `admin` and `zwave`)

**Comment by AlCalzone** [comment]
@brianmichalk1 See page 4 of the following pdf for LED and beep codes
[5069940.pdf](https://github.com/zwave-js/node-zwave-js/files/15218775/5069940.pdf)
